onnxruntime/core/providers/shared_library/provider_interfaces.h

```diff
@@ -674,7 +674,7 @@ struct ProviderHost {
 
   // GraphViewer
   virtual void GraphViewer__operator_delete(GraphViewer* p) = 0;
-  virtual std::unique_ptr<Model> GraphViewer__CreateModel(const GraphViewer* p, const logging::Logger& logger) = 0;
+  virtual std::unique_ptr<Model> GraphViewer__CreateModel(const GraphViewer* p, const logging::Logger& logger, bool is_domain_onnx_only) = 0;
```

Comment:
can you give more context about this? how/where is is_domain_onnx_only used?

Reply:
Certainly.

The modification of this interface is mainly due to the limitation of CANN EP graph parser, which requires that the "imports" attribute of the ONNX model can only include the domain "ai.onnx", while the current model parsed by onnxruntime, its "imports" attribute includes 9 domains.

There are currently two possible solutions:
1. The first is to modify the CreateModel interface. In order to ensure forward compatibility, a new default option is_onnx_domain_only is added.
2. The second is to directly construct the Model object through the Model constructor.

The location of calling this interface is as follows, the purpose is to generate a new model：
https://github.com/learningbackup/onnxruntime/blob/c4b14441ffdb1d41c5cc52f43ba57d8b15e51839/onnxruntime/core/providers/cann/cann_execution_provider.cc#L1318

Reply:
Okay. If i'm understanding correctly, only "ai.onnx" domain ONNX models are supported by CANN EP graph parser.
Are there models that you are testing which includes other domains? Most models should only include operators from "ai.onnx" domain, right?

Reply:
Sorry, I was too busy on the weekend to reply to the message in time.

Yes, CANN EP currently only supports the "ai.onnx" domain, and other domains will support it in the future

I tested 13 models before, and these models really only include the "ai.onnx" domain, and for models that include other domains, the results may be incorrect.

So I just made a new commit to fix this issue.

A brief description is as follows:
1. If the domain of the node is not "ai.onnx", then fall back to CPU
2. If the version of the "ai.onnx" domain of node is not between [8,15], fall back to CPU
3. If the op type of node is not in the operator list supported by CANN EP, it will also fall back to CPU
4. If all the inputs of node are constant inputs, fall back to CPU
5. If none of the above conditions are met, run through the CANN graph engine.

Reply:
I'm still trying to understand the change to add is_domain_onnx_only param to GraphViewer__CreateModel() 
If you filter out other domains, then you can lose portions of the model.  Shouldn't the filtering be done at the CANN EP graph parser? 
Put a different way, none of the other EP's needs the is_domain_onnx_only param. So is it only being added to work around a CANN EP limitation? I would prefer to avoid adding EP specific logic to provider shared code.

Reply:
Yes, this makes sense. Thanks for refactoring.